digdag-core/src/main/java/io/digdag/core/database/DatabaseSessionStoreManager.java

```diff
@@ -886,6 +886,7 @@ public boolean copyInitialTasksForRetry(List<Long> recursiveChildrenIdList)
             for (StoredTask task : tasks) {
                 Task newTask = Task.taskBuilder()
                     .from(task)
+                    .parentId(task.getParentId().transform(it -> oldIdToNewId.containsKey(it) ? oldIdToNewId.get(it) : it))
```

Comment:
How about simply `it -> oldIdToNewId.getOrDefault(it, it)`?

Reply:
It's better! I'll change it.